skyportal/handlers/api/image_analysis.py

```diff
@@ -0,0 +1,451 @@
+        m = pipeline.calibrate_photometry(
+            obj,
+            cat,
+            sr=2 / 3600,
+            cat_col_mag='rmag',
+            cat_col_mag1='gmag',
+            cat_col_mag2='rmag',
+            max_intrinsic_rms=0.02,
+            order=2,
+            verbose=True,
+        )
+
+        # The code above automatically augments the object list with calibrated magnitudes, but we may also do it manually
+        # obj['mag_calib'] = obj['mag'] + m['zero_fn'](obj['x'], obj['y'])
+        # obj['mag_calib_err'] = np.hypot(obj['magerr'], m['zero_fn'](obj['x'], obj['y'], get_err=True))
+
+        # ## Simple catalogue-based transient detection
+        # Some transients may already be detected by comparing the detected objects with catalogue BUT limited approach
+
+        # Filtering of transient candidates
+        candidates = pipeline.filter_transient_candidates(
+            obj, cat=cat, sr=2 / 3600, verbose=True
+        )
+        log(f'{candidates}')
+
+        # Creating cutouts for these candidates and vizualizing them (6238 ???)
+        filtered = []
+        for i, cand in enumerate(candidates):
+            dist = spherical_distance(
+                candidates[i]['ra'], candidates[i]['dec'], 191.4734, 89.1846
+            )  # XRT enhanced position
+            if dist < 4 / 3600:
+                filtered.append(candidates[i])
+
+        for i, cand in enumerate(filtered):
+            # Create the cutout from image based on the candidate
+            cutout = cutouts.get_cutout(image, cand, 20, mask=mask, header=header)
+            # We may directly download the template image for this cutout from HiPS server - same scale and orientation
+            cutout['template'] = templates.get_hips_image(
+                'PanSTARRS/DR1/r', header=cutout['header']
+            )[0]
+            # We do not have difference image, so it will only display original one, template and mask
+            plots.plot_cutout(cutout, qq=[0.5, 99.9], stretch='linear')
+
+        # compute upper limit
+        ra0, dec0, sr0 = astrometry.get_frame_center(
+            wcs=wcs, width=image.shape[1], height=image.shape[0]
+        )
+        pixscale = astrometry.get_pixscale(wcs=wcs)
+
+        zero_fn = m[
+            'zero_fn'
+        ]  # Function to get the zero point as a function of position on the image
+        obj['mag_calib'] = obj['mag'] + zero_fn(obj['x'], obj['y'])
+
+        # We may roughly estimage the effective gain of the image from background mean and rms as gain = mean/rms**2
+        bg, rms = photometry.get_background(image, mask=mask, get_rms=True)
+
+        log(f'Effective gain is {np.median(bg/rms**2)}')
+
+        psf_model, psf_snapshots = psf.run_psfex(
+            image,
+            mask=mask,
+            checkimages=['SNAPSHOTS'],
+            order=0,
+            verbose=True,
+            _workdir=workdir_psfex,
+        )
+
+        sims = []
```

Comment:
Add a comment explaining why we are adding simulations here.

Reply:
@Theodlz See above. It is to make a limiting magnitude assessment.